src/dex/woo-fi/woo-fi.ts

```diff
@@ -0,0 +1,556 @@
+      return `${this.dexKey.toLowerCase()}_qb`;
+    } else if (destToken.toLowerCase() === this.quoteTokenAddress) {
+      return `${this.dexKey.toLowerCase()}_bq`;
+    } else {
+      throw new Error(
+        'srcToken or destToken must be equal to quoteTokenAddress',
+      );
+    }
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    const _srcToken = wrapETH(srcToken, this.network);
+    const _destToken = wrapETH(destToken, this.network);
+
+    if (_srcToken.address.toLowerCase() === _destToken.address.toLowerCase()) {
+      return [];
+    }
+
+    let tokenToSearch: string;
+    if (_srcToken.address.toLowerCase() === this.quoteTokenAddress) {
+      tokenToSearch = _destToken.address.toLowerCase();
+    } else if (_destToken.address.toLowerCase() === this.quoteTokenAddress) {
+      tokenToSearch = _srcToken.address.toLowerCase();
+    } else {
+      return [];
+    }
+    const baseToken = this.baseTokens.find(
+      token => token.address.toLowerCase() === tokenToSearch,
+    );
+    return baseToken === undefined
+      ? []
+      : [this.getIdentifier(_srcToken.address, _destToken.address)];
+  }
+
+  getBaseFromIdentifier(
+    identifier: string,
+    srcToken: Address,
+    destToken: Address,
+  ) {
+    if (this.tokenByAddress === null)
+      throw new Error(
+        'tokenByAddress was not properly initialized. Check if initializePricing was called',
+      );
+
+    const direction = identifier.split('_')[1];
+    if (direction === 'bq') {
+      return this.tokenByAddress[srcToken.toLowerCase()];
+    } else if (direction === 'qb') {
+      return this.tokenByAddress[destToken.toLowerCase()];
+    } else {
+      throw new Error(
+        `direction in getBaseFromIdentifier must be 'bq' or 'qb'`,
+      );
+    }
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<WooFiData>> {
+    if (side === SwapSide.BUY) return null;
+
+    try {
+      const _srcToken = wrapETH(srcToken, this.network);
+      const _destToken = wrapETH(destToken, this.network);
+
+      if (
+        _srcToken.address.toLowerCase() === _destToken.address.toLowerCase()
+      ) {
+        return null;
+      }
+
+      const expectedIdentifier = this.getIdentifier(
+        _srcToken.address,
+        _destToken.address,
+      );
+      const allowedPairIdentifiers =
+        limitPools !== undefined
+          ? limitPools.filter(
+              limitIdentifier => limitIdentifier === expectedIdentifier,
+            )
+          : await this.getPoolIdentifiers(
+              _srcToken,
+              _destToken,
+              side,
+              blockNumber,
+            );
+
+      if (!allowedPairIdentifiers.length) return null;
```

Comment:
This can be also written much more efficiently 
```suggestion
try {
      if(side === SwapSide.BUY) return null;
      
      const _srcToken = wrapETH(srcToken, this.network);
      const _destToken = wrapETH(destToken, this.network);
      const _srcAddress = _srcToken.toLowerCase();
      const _destAddress = _destToken.toLowerCase();
      
      if(!this.tokenByAddress[_srcAddress] || !this.tokenByAddress[_destAddress])
       return null;
       
      const isSrcQuote = _srcAddress === this.quoteTokenAddress;
      const isDestQuote = _destAddress === this.quoteTokenAddress;
      if(!isSrcQuote  && !isDestQuote)
         return null;
    
      const expectedIdentifier = this.getIdentifier(_srcAddress, _destAddress);
        if(!limitPools.some(p => p === expectedIdentifier))
            return null;
```